### ISSUE TITLE
Unable to select anything in `system.parts` using UUID

### ISSUE THREAD

**Issue by Selfeer** [ClickHouse/ClickHouse#61220]
When trying to select a table name or a part name from `system.parts` table using `uuid` we don't get any output, starting from ClickHouse 24.

For example,

```sql
CREATE TABLE users (uid Int16, name String, age Int16) ENGINE=MergeTree ORDER BY uid PARTITION BY uid;

INSERT INTO users VALUES (1231, 'John', 33);
INSERT INTO users VALUES (6666, 'Ksenia', 48);

SELECT uuid, name from system.parts WHERE table = 'users';

SELECT table from system.parts WHERE uuid = '00000000-0000-0000-0000-000000000000';
``` 

This query in 23.12 would output expected

```sql
00000000-0000-0000-0000-000000000000	1231_1_1_0
00000000-0000-0000-0000-000000000000	6666_2_2_0
users
users
```

But in >=24.1 we only get the output from the 

```sql
SELECT uuid, name from system.parts WHERE table = 'users';

00000000-0000-0000-0000-000000000000	1231_1_1_0
00000000-0000-0000-0000-000000000000	6666_2_2_0
```

and the following query just returns an empty value, when it should return the name of the table which the part belongs to and has the given uuid:

```sql
SELECT table from system.parts WHERE uuid = '00000000-0000-0000-0000-000000000000';
```

Here's the example for 24.2: https://fiddle.clickhouse.com/e779eee6-48fc-4cac-b16a-9d58e0700d24
Here's how it was behaving before the 24 release: https://fiddle.clickhouse.com/c07dc612-876f-4b15-bef7-8ee63ae68ee3

**Comment by wudanzy** [comment]
I reproduced this in my local machine, trying to investigate.

**Comment by wudanzy** [comment]
I found it works when using the Analyzer, could you please confirm that?

To enable Analyzer, please set the following:
```
SET allow_experimental_analyzer=1;
```

**Comment by wudanzy** [comment]
One quick update is that the root cause is in the storage layer (instead of filtering logic), will continue investigating later.

**Comment by wudanzy** [comment]
I have a hypothesis for the root cause of this problem, as follows:
1. The storage layer tried to do the filtering on the condition, so the AST (which contains the predicate) is passed down to `StorageSystemPartsBase`.
2. `StorageSystemPartsBase` used this predicate to do filtering, when choosing storages.
3. However, storage also has their UUID (I am new to Clickhouse, but I guess storage UUID and parts UUID are different thing), so this predicate filters out all storages since their UUID are not `00000000-0000-0000-0000-000000000000`.
4. There's no result returned.

In the Analyzer path and non-analyzer path, the AST is different:
Non-analyzer path:
```
SelectQuery, 0x00007f84915a1618
-ExpressionList, 0x00007f849156c718
--Identifier_uuid, 0x00007f84915c9b98
--Identifier_name, 0x00007f84915c9cd8
-TablesInSelectQuery, 0x00007f84915746d8
--TablesInSelectQueryElement, 0x00007f84915fe518
---TableExpression, 0x00007f84915c9198
----TableIdentifier_system.parts, 0x00007f84915c92d8
-Function_equals, 0x00007f8491578398
--ExpressionList, 0x00007f8491570b58
---Identifier_uuid, 0x00007f84915c8798
---Literal_'00000000-0000-0000-0000-000000000000', 0x00007f8491577918
```
Analyzer path:
```
SelectQuery, 0x00007f848b327218
-ExpressionList, 0x00007f848b3e23b8
--Identifier___table1.uuid, 0x00007f84915c9418
--Identifier___table1.name, 0x00007f84915c8798
-TablesInSelectQuery, 0x00007f848b3ba018
--TablesInSelectQueryElement, 0x00007f848b327318
---TableExpression, 0x00007f84915c8b58
----TableIdentifier_system.parts, 0x00007f84915c9198
-Function_equals, 0x00007f8491577a98
--ExpressionList, 0x00007f848b3df1d8
---Identifier___table1.uuid, 0x00007f84915c8658
---Literal_'00000000-0000-0000-0000-000000000000', 0x00007f8491577798
```

Since an internal temporal table contains `uuid` column, so in non-analyzer path, it extracted a predicate on this column, but in analyzer path, it didn't think such predicate would match any column.

So another trick to show the result is change the predicate to `uuid != '00000000-0000-0000-0000-000000000001'`. In this case, it avoid filtering out all storages, so we can see the result.

**Comment by wudanzy** [comment]
I see the code changed a lot, I will check if the newest code still have such a problem and fix it.

**Comment by wudanzy** [comment]
I think the bug is introduced by [this PR](https://github.com/ClickHouse/ClickHouse/commit/3d8933f2410687d43d8e5d5b8d538b4c79c4f740#diff-6b38d363b27bedd99c449eb3b8b4a2bb3b271d4313659f2e050bc3cf54e1efe2), where it adds a UUID column. It became a bug because the name `uuid` happens to conflicts with a field in system.parts.

A simple fix is to rename `uuid` as `storage_uuid`. This is totally fine because this name is only used in StoragesInfoStream::StoragesInfoStream(), and will be discarded right away. Also it is impossible to hack this by specify `WHERE storage_uuid = '00000000-0000-0000-0000-000000000000' `, because analyzer would reject this since there is no column called `storage_uuid` in `system.parts`.

**Comment by antaljanosbenjamin** [comment]
Fixed by #61479.
